### PR TITLE
631 fix os proc pool eunit

### PR DESCRIPTION
**Pull request by davisp**
This is a fix for #631. Basically just replacse the couchjs processes with couch_native_proc instances since we're just worried about how the proc manager behaves rather than anything to do with JavaScript.

### CODE REVIEW

File: src/couch/src/couch_native_process.erl [hunk 1]
```diff
@@ -65,10 +72,12 @@ prompt(Pid, Data) when is_list(Data) ->
 
 % gen_server callbacks
 init([]) ->
-    {ok, #evstate{ddocs=dict:new()}}.
+    V = config:get("query_server_config", "os_process_idle_limit", "300"),
```
Comment (wohali) [inline src/couch/src/couch_native_process.erl]: Please add this new default value to `rel/overlay/default.ini`.

File: src/couch/src/couch_native_process.erl [hunk 2]
```diff
@@ -65,10 +72,12 @@ prompt(Pid, Data) when is_list(Data) ->
...
+    Idle = list_to_integer(V) * 500,
```
Comment (wohali) [inline src/couch/src/couch_native_process.erl]: Why 500? If it's to convert to ms wouldn't it be 1000?

Comment (davisp) [inline src/couch/src/couch_native_process.erl]: Ooops! Was debugging why timeouts weren't working and forgot to undo that debug change. squashvertipushing.

File: src/couch/src/couch_native_process.erl [hunk 3]
```diff
@@ -79,28 +88,38 @@ handle_call({prompt, Data}, _From, State) ->
...
+    {noreply, State, State#evstate.idle};
+handle_cast(stop, State) ->
+    {stop, normal, State};
+handle_cast(_Msg, State) ->
+    {noreply, State, State#evstate.idle}.
 
-handle_info({'EXIT',_,normal}, State) -> {noreply, State};
+handle_info(timeout, State) ->
+    gen_server:cast(couch_proc_manager, {os_proc_idle, self()}),
+    erlang:garbage_collect(),
```
Comment (wohali) [inline src/couch/src/couch_native_process.erl]: No specific objection, but is there a specific reason we want to garbage collect after a process times out?

Comment (nickva) [inline src/couch/src/couch_native_process.erl]: This is what we do in the os_process as well

**Comment by nickva** [comment]
+1 (with the default.ini change Wohali mentioned)

Also here is a longer 10-build test run of the same: 

https://travis-ci.org/cloudant/couchdb/builds/252941547?utm_source=github_status&utm_medium=notification

**Comment by davisp** [comment]
@nickva What default.ini change?

File: src/couch/src/couch_native_process.erl (hunk 1 above)
Comment (davisp) [inline src/couch/src/couch_native_process.erl]: Can add, but just wrote this to match couch_os_process, its not new.

**Comment by davisp** [comment]
Ah, page wasn't refreshed. Will add the default as a new commit for the history.

File: src/couch/src/couch_native_process.erl (hunk 1 above)
Comment (wohali) [inline src/couch/src/couch_native_process.erl]: Never mind, I missed that it was there (but commented out).

Comment (davisp) [inline src/couch/src/couch_native_process.erl]: Note that we all just realized in IRC that the default is already there.

**Comment by nickva** [comment]
@davisp sorry, meant adding os_process_idle_limit to default.ini, but I just checked and it's already there just commented out.

**Comment by davisp** [comment]
Calling out my comment that the request change to add os_process_idle_limit is unneeded since its already there. Just typoed a grep so merging this as is.

**Pull request merged**